pkg/io/option.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io
+
+import (
+	"cloud.google.com/go/storage"
+	"gocloud.dev/blob"
+	"google.golang.org/api/googleapi"
+	utilpointer "k8s.io/utils/pointer"
+)
+
+// WriterOptions are options for the opener Writer method
+type WriterOptions struct {
+	BufferSize      *int64
+	ContentEncoding *string
+	ContentType     *string
+	Metadata        *map[string]string
+}
+
+// Apply applies the WriterOptions to storage.Writer and blob.WriterOptions
+// Both arguments are allowed to be nil.
+func (wo WriterOptions) Apply(writer *storage.Writer, o *blob.WriterOptions) {
+	if writer != nil {
+		if wo.BufferSize != nil {
+			if *wo.BufferSize < googleapi.DefaultUploadChunkSize {
+				writer.ChunkSize = int(*wo.BufferSize)
+			}
+		}
+		if wo.ContentEncoding != nil {
+			writer.ObjectAttrs.ContentEncoding = *wo.ContentEncoding
+		}
+		if wo.ContentType != nil {
+			writer.ObjectAttrs.ContentType = *wo.ContentType
+		}
+		if wo.Metadata != nil {
+			writer.ObjectAttrs.Metadata = *wo.Metadata
+		}
+	}
```

Comment:
return? Then you don't need the indent below

Reply:
I'm assuming you mean it like this?
```go
if o == nil {
   return 
}
... (everything else without indent)
```